README.md

```diff
@@ -1,16 +1,110 @@
-## exclusive-lock-node
+<!-- TOC -->
 
+- [`ExclusiveLock`](#exclusivelock)
+- [Installation](#installation)
+- [Usage](#usage)
+- [API](#api)
+  - [`new ExclusiveLock(options)`](#new-exclusivelockoptions)
+  - [`acquire()`](#acquire)
+  - [`inspect()`](#inspect)
+  - [`release()`](#release)
+  - [`refresh()`](#refresh)
+- [Authors](#authors)
+- [License](#license)
 
+<!-- /TOC -->
+## `ExclusiveLock`
+
+Need to synchronize a bunch of workers so that only 1 does a particular thing?  If yes,
+then you need a lock file.  This package uses a cache store (`redis` or `keydb`)
+to create records atomically that only 1 instance will "own."
 ## Installation
 
+```bash
+npm install @logdna/exclusive-lock --save
+```
 
 ## Usage
 
+```js
+const ExclusiveLock = require('@logdna/exclusive-lock')
+const cache_connection = require('./my-keydb-connection.js')
+const pino = require('pino')
+
+const exclusive_lock = new ExclusiveLock({
+  app_name: 'my-distributed-app'
+, log: pino()
+, cache_manager
+})
+
+async function main() {
+  const got_lock = await exclusive_lock.acquire()
+  if (got_lock) {
+    log.info('You win the race! Lock acquired')
+  }
+  exclusive_lock.release()
+}
+
+main()
+```
+
+## API
```

Comment:
Good one!

We could add more info about functionality and limitations, something like:

> NOTES
> This library covers most common use cases for exclusion when working with multiple instances of the same application / container: 
> - Exclusive initialization: have one of the application instances perform a task on init.
> - Periodic checks: Have one of the application instances perform a task periodically.

> For long running tasks and rare edge cases, it could be possible that the lock is perceived as acquired by more than one instance. This library does not provide strict guarantees for exclusion for tasks that must be processed exactly once in a non-idempotent manner (e.g. appending an item in order processing). Consider using upserts/cas on your persistent DB in those cases or using a dedicated solution for exclusion like etcd / zookeeper.